src/spatialdata/_core/spatialdata.py

```diff
@@ -962,195 +908,564 @@ def transform_to_coordinate_system(
                 elements[element_type][element_name] = transformed
         return SpatialData(**elements, tables=sdata.tables)
 
-    def write(
+    def elements_are_self_contained(self) -> dict[str, bool]:
+        """
+        Describe if elements are self-contained as a dict of element_name to bool.
+
+        Returns
+        -------
+        A dictionary of element_name to boolean values indicating whether the elements are self-contained.
+        """
+        from spatialdata._io._utils import _is_element_self_contained
+
+        if self.path is None:
+            return {element_name: True for _, element_name, _ in self.gen_elements()}
+
+        self._validate_element_names_are_unique()
+        description = {}
+        for element_type, element_name, element in self.gen_elements():
+            element_path = self.path / element_type / element_name
+            description[element_name] = _is_element_self_contained(element, element_path)
+        return description
+
+    def is_self_contained(self, element_name: str | None = None) -> bool:
+        """
+        Check if an object is self-contained; self-contained objects have a simpler disk storage layout.
+
+        A SpatialData object is said to be self-contained if all its SpatialElements or AnnData tables are
+        self-contained. A SpatialElement or AnnData table is said to be self-contained when it does not depend on a
+        Dask computational graph (i.e. it is not "lazy") or when it is Dask-backed and each file that is read in the
+        Dask computational graph is contained within the Zarr store associated with the SpatialElement.
```

Comment:
could this description fo self-containing also be in the `elements_are_self_contained` above?